### PR TITLE
Implemented basic interaction interface and tests.

### PR DESCRIPTION
**Pull request by joguSD**
Slight wizards refactor to facilitate the integration of interactions.

* Interactions are now mocked for testing.
* Basic interface for interactions implemented, integrated, and tested.
* Can display a simple selector
* Can select from a few options in python

Unit test coverage is now up to 100% for wizard functionality.

### CODE REVIEW

File: awsshell/interaction.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    def __init__(self, model, prompt_msg, prompter=prompt):
+        super(SimpleSelect, self).__init__(model, prompt_msg)
+        self._prompter = prompter
+
+    def execute(self, data):
+        if not isinstance(data, list) or len(data) < 1:
+            raise InteractionException('SimpleSelect expects a non-empty list')
+        if self._model.get('Path') is not None:
+            disp_data = jmespath.search(self._model['Path'], data)
+            cmpltr = WordCompleter(disp_data, sentence=True, ignore_case=True)
```
Comment (JordonPhillips) [inline awsshell/interaction.py]: I'd prefer variable names to not be shortened unless it's something like stdout that's super common.

File: awsshell/wizard.py [hunk 2]
```diff
@@ -18,17 +19,25 @@ class WizardLoader(object):
...
 
-    def __init__(self, session=None):
+    def __init__(self, session=None, interaction_loader=None):
         """Initialize a wizard factory.
 
         :type session: :class:`botocore.session.Session`
         :param session: (Optional) The botocore session to be used when loading
         models and retrieving clients.
+
+        :type session: :class:`awsshell.interaction.InteractionLoader`
```
Comment (JordonPhillips) [inline awsshell/wizard.py]: s/session/interaction_loader

**Comment by JordonPhillips** [comment]
Looks good! Just had a few small comments.

File: awsshell/interaction.py (hunk 1 above)
Comment (joguSD) [inline awsshell/interaction.py]: I did this to get the line under 79 characters. I went back and forth on this but I think I can agree it's more readable as the whole word. I'll refactor this so it's broken up on multiple lines.

**Comment by JordonPhillips** [comment]
⛵

**Pull request merged**